src/rules/noLookaheadLookbehindRegex.ts

```diff
@@ -85,8 +85,25 @@ const noLookaheadLookbehindRegexp: Rule.RuleModule = {
     if (!unsupportedTargets.length && hasConfig) return {};
 
     return {
+      TemplateLiteral(node: ESTree.TemplateLiteral & Rule.NodeParentExtension): void {
+        for (let quasi of node.quasis) {
```

Comment:
I think this would end up reporting the problem on any string template literal (regardless if it is used inside a regexp or not).  We should first check if the template literal is inside a regexp constructor before we iterate over all the nodes